### ISSUE TITLE
FreshClam received error code 403 from the ClamAV Content Delivery Network (CDN).

### ISSUE THREAD

**Issue by mks-mr** [Cisco-Talos/clamav#851]
I am unable to download databases and end up being put on "cool-down" all the time.

OS: Debian Bullseye
ClamAV Version: 0.103.8
IP: 91.107.225.157

Same version works for all other machines I'm running.

Output below.

Wed Mar  1 18:14:21 2023 -> ClamAV update process started at Wed Mar  1 18:14:21 2023
Wed Mar  1 18:14:21 2023 -> WARNING: Cool-down expired, ok to try again.
Wed Mar  1 18:14:21 2023 -> daily database available for update (local version: 26820, remote version: 26827)
Wed Mar  1 18:14:23 2023 -> WARNING: downloadPatch: Can't download daily-26821.cdiff from https://database.clamav.net/daily-26821.cdiff
Wed Mar  1 18:14:23 2023 -> WARNING: Incremental update failed, trying to download daily.cvd
Wed Mar  1 18:14:23 2023 -> WARNING: Can't download daily.cvd from https://database.clamav.net/daily.cvd
Wed Mar  1 18:14:23 2023 -> WARNING: FreshClam received error code 403 from the ClamAV Content Delivery Network (CDN).
Wed Mar  1 18:14:23 2023 -> This could mean several things:
Wed Mar  1 18:14:23 2023 ->  1. You are running an out-of-date version of ClamAV / FreshClam.
Wed Mar  1 18:14:23 2023 ->     Ensure you are the most updated version by visiting https://www.clamav.net/downloads
Wed Mar  1 18:14:23 2023 ->  2. Your network is explicitly denied by the FreshClam CDN.
Wed Mar  1 18:14:23 2023 ->     In order to rectify this please check that you are:
Wed Mar  1 18:14:23 2023 ->    a. Running an up-to-date version of FreshClam
Wed Mar  1 18:14:23 2023 ->    b. Running FreshClam no more than once an hour
Wed Mar  1 18:14:23 2023 ->    c. If you have checked (a) and (b), please open a ticket at
Wed Mar  1 18:14:23 2023 ->       https://github.com/Cisco-Talos/clamav/issues
Wed Mar  1 18:14:23 2023 ->       and we will investigate why your network is blocked.
Wed Mar  1 18:14:23 2023 -> WARNING: You are on cool-down until after: 2023-03-02 18:14:23
Wed Mar  1 18:14:23 2023 -> ERROR: Database update process failed: Forbidden; Blocked by CDN
Wed Mar  1 18:14:23 2023 -> ERROR: Update failed.

**Comment by micahsnyder** [comment]
Can I trouble you to delete the `freshclam.dat` file from the database directory and then run `freshclam` with `--verbose` and gather the results.  The HTTP response should include a RayID value that will help us identify in case searching by your IP is insufficient.

**Comment by mks-mr** [comment]
Hello Micah,

done.

Here's what I get:

# freshclam --verbose
Thu Mar  2 09:26:45 2023 -> *Current working dir is /var/lib/clamav/
Thu Mar  2 09:26:45 2023 -> *Can't open freshclam.dat in /var/lib/clamav
Thu Mar  2 09:26:45 2023 -> *It probably doesn't exist yet. That's ok.
Thu Mar  2 09:26:45 2023 -> *Failed to load freshclam.dat; will create a new freshclam.dat
Thu Mar  2 09:26:45 2023 -> *Creating new freshclam.dat
Thu Mar  2 09:26:45 2023 -> *Saved freshclam.dat
Thu Mar  2 09:26:45 2023 -> ClamAV update process started at Thu Mar  2 09:26:45 2023
Thu Mar  2 09:26:45 2023 -> *Current working dir is /var/lib/clamav/
Thu Mar  2 09:26:45 2023 -> *Querying current.cvd.clamav.net
Thu Mar  2 09:26:45 2023 -> *TTL: 185
Thu Mar  2 09:26:45 2023 -> *fc_dns_query_update_info: Software version from DNS: 0.103.8
Thu Mar  2 09:26:45 2023 -> *Current working dir is /var/lib/clamav/
Thu Mar  2 09:26:45 2023 -> *check_for_new_database_version: Local copy of daily found: daily.cvd.
Thu Mar  2 09:26:45 2023 -> *query_remote_database_version: daily.cvd version from DNS: 26828
Thu Mar  2 09:26:45 2023 -> daily database available for update (local version: 26820, remote version: 26828)
Current database is 8 versions behind.
Downloading database patch # 26821...
Thu Mar  2 09:26:46 2023 -> *Retrieving https://database.clamav.net/daily-26821.cdiff
Thu Mar  2 09:26:46 2023 -> *downloadFile: Download source:      https://database.clamav.net/daily-26821.cdiff
Thu Mar  2 09:26:46 2023 -> *downloadFile: Download destination: ./clamav-b4412a704dca2f7b42f97f3b4a91995b.tmp
*   Trying 104.16.218.84:443...
* Connected to database.clamav.net (104.16.218.84) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=Cloudflare, Inc.; CN=sni.cloudflaressl.com
*  start date: Jun 14 00:00:00 2022 GMT
*  expire date: Jun 14 23:59:59 2023 GMT
*  subjectAltName: host "database.clamav.net" matched cert's "database.clamav.net"
*  issuer: C=US; O=Cloudflare, Inc.; CN=Cloudflare Inc ECC CA-3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x5638034bdfe0)
> GET /daily-26821.cdiff HTTP/2
Host: database.clamav.net
user-agent: ClamAV/0.103.8 (OS: linux-gnu, ARCH: x86_64, CPU: x86_64, UUID: 4f96dba5-f2ad-4095-a840-f27aa8e606d0)
accept: */*
connection: close

* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 256)!
< HTTP/2 403 
< date: Thu, 02 Mar 2023 08:26:46 GMT
< content-type: text/plain; charset=UTF-8
< content-length: 16
< x-frame-options: SAMEORIGIN
< referrer-policy: same-origin
< cache-control: private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0
< expires: Thu, 01 Jan 1970 00:00:01 GMT
< server-timing: cf-q-config;dur=6.0000002122251e-06
< strict-transport-security: max-age=15552000
< x-content-type-options: nosniff
< server: cloudflare
< cf-ray: 7a1841ba8efcc2de-VIE
< 
Time:    0.1s, ETA:    0.0s [========================>]        16B/16B
* Connection #0 to host database.clamav.net left intact
Thu Mar  2 09:26:46 2023 -> *Saved freshclam.dat
Thu Mar  2 09:26:46 2023 -> ^downloadPatch: Can't download daily-26821.cdiff from https://database.clamav.net/daily-26821.cdiff
Thu Mar  2 09:26:46 2023 -> ^Incremental update failed, trying to download daily.cvd
Thu Mar  2 09:26:46 2023 -> *Retrieving https://database.clamav.net/daily.cvd
Thu Mar  2 09:26:46 2023 -> *downloadFile: Download source:      https://database.clamav.net/daily.cvd
Thu Mar  2 09:26:46 2023 -> *downloadFile: Download destination: /var/lib/clamav/tmp.7a09b31573/clamav-add82f41e044953d107453b90d7a4cae.tmp
*   Trying 104.16.218.84:443...
* Connected to database.clamav.net (104.16.218.84) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=Cloudflare, Inc.; CN=sni.cloudflaressl.com
*  start date: Jun 14 00:00:00 2022 GMT
*  expire date: Jun 14 23:59:59 2023 GMT
*  subjectAltName: host "database.clamav.net" matched cert's "database.clamav.net"
*  issuer: C=US; O=Cloudflare, Inc.; CN=Cloudflare Inc ECC CA-3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x563803555ec0)
> GET /daily.cvd HTTP/2
Host: database.clamav.net
user-agent: ClamAV/0.103.8 (OS: linux-gnu, ARCH: x86_64, CPU: x86_64, UUID: 4f96dba5-f2ad-4095-a840-f27aa8e606d0)
accept: */*
if-modified-since: Wed, 22 Feb 2023 08:30:58 GMT
connection: close

* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 256)!
< HTTP/2 403 
< date: Thu, 02 Mar 2023 08:26:46 GMT
< content-type: text/plain; charset=UTF-8
< content-length: 16
< x-frame-options: SAMEORIGIN
< referrer-policy: same-origin
< cache-control: private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0
< expires: Thu, 01 Jan 1970 00:00:01 GMT
< server-timing: cf-q-config;dur=7.0000005507609e-06
< strict-transport-security: max-age=15552000
< x-content-type-options: nosniff
< server: cloudflare
< cf-ray: 7a1841bae89c788b-VIE
< 
Time:    0.1s, ETA:    0.0s [========================>]        16B/16B
* Connection #0 to host database.clamav.net left intact
Thu Mar  2 09:26:46 2023 -> *Saved freshclam.dat
Thu Mar  2 09:26:46 2023 -> ^Can't download daily.cvd from https://database.clamav.net/daily.cvd
Thu Mar  2 09:26:46 2023 -> ^FreshClam received error code 403 from the ClamAV Content Delivery Network (CDN).
Thu Mar  2 09:26:46 2023 -> This could mean several things:
Thu Mar  2 09:26:46 2023 ->  1. You are running an out-of-date version of ClamAV / FreshClam.
Thu Mar  2 09:26:46 2023 ->     Ensure you are the most updated version by visiting https://www.clamav.net/downloads
Thu Mar  2 09:26:46 2023 ->  2. Your network is explicitly denied by the FreshClam CDN.
Thu Mar  2 09:26:46 2023 ->     In order to rectify this please check that you are:
Thu Mar  2 09:26:46 2023 ->    a. Running an up-to-date version of FreshClam
Thu Mar  2 09:26:46 2023 ->    b. Running FreshClam no more than once an hour
Thu Mar  2 09:26:46 2023 ->    c. If you have checked (a) and (b), please open a ticket at
Thu Mar  2 09:26:46 2023 ->       https://github.com/Cisco-Talos/clamav/issues
Thu Mar  2 09:26:46 2023 ->       and we will investigate why your network is blocked.
Thu Mar  2 09:26:46 2023 -> ^You are on cool-down until after: 2023-03-03 09:26:46
Thu Mar  2 09:26:46 2023 -> !Database update process failed: Forbidden; Blocked by CDN
Thu Mar  2 09:26:46 2023 -> !Update failed.
--- End ---

One failed attempt and I'm getting blocked for 24 hrs, duh...

**Comment by micahsnyder** [comment]
Hi @mks-mr,

The message "You are on cool-down until after..." is self-imposed by Freshclam when it receives a 403 or 429 to reduce the amount of 403/429's sent by spamming clients.  I believe our Cloudflare configuration blocks for 12 hours after rate limiting starts.

But in this case our Cloudflare admin wasn't able to find any record of you hitting our firewall rules in the last 72 hours (as far back as the data goes).  Also, rate limiting should be a 429 response (not a 403), so I don't think it is that.  

There are a handful of reasons why Cloudflare would serve up a 403 response before it ever gets to our rules.  They are:

![image](https://user-images.githubusercontent.com/30635813/222499082-ddd96e83-5e78-41cc-b871-59528993d0b8.png)

I'm not sure if any of the above apply.  It doesn't really make sense to me.  We just tried adding your public IP to an allow list to see if that helps.  If Cloudflare is blocking you before it gets to us, then it may not.   Please delete that `freshclam.dat` file and try again and let me know how it goes. 

Regards,
Micah

**Comment by mks-mr** [comment]
No success. The output is:

Mon Mar  6 14:46:27 2023 -> *Current working dir is /var/lib/clamav/
Mon Mar  6 14:46:27 2023 -> *Can't open freshclam.dat in /var/lib/clamav
Mon Mar  6 14:46:27 2023 -> *It probably doesn't exist yet. That's ok.
Mon Mar  6 14:46:27 2023 -> *Failed to load freshclam.dat; will create a new freshclam.dat
Mon Mar  6 14:46:27 2023 -> *Creating new freshclam.dat
Mon Mar  6 14:46:27 2023 -> *Saved freshclam.dat
Mon Mar  6 14:46:27 2023 -> ClamAV update process started at Mon Mar  6 14:46:27 2023
Mon Mar  6 14:46:27 2023 -> *Current working dir is /var/lib/clamav/
Mon Mar  6 14:46:27 2023 -> *Querying current.cvd.clamav.net
Mon Mar  6 14:46:27 2023 -> *TTL: 1297
Mon Mar  6 14:46:27 2023 -> *fc_dns_query_update_info: Software version from DNS: 0.103.8
Mon Mar  6 14:46:27 2023 -> *Current working dir is /var/lib/clamav/
Mon Mar  6 14:46:27 2023 -> *check_for_new_database_version: Local copy of daily found: daily.cvd.
Mon Mar  6 14:46:27 2023 -> *query_remote_database_version: daily.cvd version from DNS: 26833
Mon Mar  6 14:46:27 2023 -> daily database available for update (local version: 26820, remote version: 26833)
Mon Mar  6 14:46:28 2023 -> *Retrieving https://database.clamav.net/daily-26821.cdiff
Mon Mar  6 14:46:28 2023 -> *downloadFile: Download source:      https://database.clamav.net/daily-26821.cdiff
Mon Mar  6 14:46:28 2023 -> *downloadFile: Download destination: ./clamav-612eae09cb9359b2b90836ac72233225.tmp
*   Trying 104.16.219.84:443...
* Connected to database.clamav.net (104.16.219.84) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=Cloudflare, Inc.; CN=sni.cloudflaressl.com
*  start date: Jun 14 00:00:00 2022 GMT
*  expire date: Jun 14 23:59:59 2023 GMT
*  subjectAltName: host "database.clamav.net" matched cert's "database.clamav.net"
*  issuer: C=US; O=Cloudflare, Inc.; CN=Cloudflare Inc ECC CA-3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x55694d044ff0)
> GET /daily-26821.cdiff HTTP/2
Host: database.clamav.net
user-agent: ClamAV/0.103.8 (OS: linux-gnu, ARCH: x86_64, CPU: x86_64, UUID: 7620ebca-7bc8-4363-b3fc-5a6f76084442)
accept: */*
connection: close

* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 256)!
< HTTP/2 403 
< date: Mon, 06 Mar 2023 13:46:29 GMT
< content-type: text/plain; charset=UTF-8
< content-length: 16
< x-frame-options: SAMEORIGIN
< referrer-policy: same-origin
< cache-control: private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0
< expires: Thu, 01 Jan 1970 00:00:01 GMT
< server-timing: cf-q-config;dur=7.9999881563708e-06
< strict-transport-security: max-age=15552000
< x-content-type-options: nosniff
< server: cloudflare
< cf-ray: 7a3b0b8b693bc25e-VIE
< 
* Connection #0 to host database.clamav.net left intact
Mon Mar  6 14:46:29 2023 -> *Saved freshclam.dat
Mon Mar  6 14:46:29 2023 -> ^downloadPatch: Can't download daily-26821.cdiff from https://database.clamav.net/daily-26821.cdiff
Mon Mar  6 14:46:29 2023 -> ^Incremental update failed, trying to download daily.cvd
Mon Mar  6 14:46:29 2023 -> *Retrieving https://database.clamav.net/daily.cvd
Mon Mar  6 14:46:29 2023 -> *downloadFile: Download source:      https://database.clamav.net/daily.cvd
Mon Mar  6 14:46:29 2023 -> *downloadFile: Download destination: /var/lib/clamav/tmp.79f794e926/clamav-b9020068829214334dbbef1e61f2b462.tmp
*   Trying 104.16.219.84:443...
* Connected to database.clamav.net (104.16.219.84) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=Cloudflare, Inc.; CN=sni.cloudflaressl.com
*  start date: Jun 14 00:00:00 2022 GMT
*  expire date: Jun 14 23:59:59 2023 GMT
*  subjectAltName: host "database.clamav.net" matched cert's "database.clamav.net"
*  issuer: C=US; O=Cloudflare, Inc.; CN=Cloudflare Inc ECC CA-3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x55694d0dcae0)
> GET /daily.cvd HTTP/2
Host: database.clamav.net
user-agent: ClamAV/0.103.8 (OS: linux-gnu, ARCH: x86_64, CPU: x86_64, UUID: 7620ebca-7bc8-4363-b3fc-5a6f76084442)
accept: */*
if-modified-since: Wed, 22 Feb 2023 08:30:58 GMT
connection: close

* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 256)!
< HTTP/2 403 
< date: Mon, 06 Mar 2023 13:46:29 GMT
< content-type: text/plain; charset=UTF-8
< content-length: 16
< x-frame-options: SAMEORIGIN
< referrer-policy: same-origin
< cache-control: private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0
< expires: Thu, 01 Jan 1970 00:00:01 GMT
< server-timing: cf-q-config;dur=6.9999950937927e-06
< strict-transport-security: max-age=15552000
< x-content-type-options: nosniff
< server: cloudflare
< cf-ray: 7a3b0b8bfbb0c2a7-VIE
< 
* Connection #0 to host database.clamav.net left intact
Mon Mar  6 14:46:29 2023 -> *Saved freshclam.dat
Mon Mar  6 14:46:29 2023 -> ^Can't download daily.cvd from https://database.clamav.net/daily.cvd
Mon Mar  6 14:46:29 2023 -> ^FreshClam received error code 403 from the ClamAV Content Delivery Network (CDN).
Mon Mar  6 14:46:29 2023 -> This could mean several things:
Mon Mar  6 14:46:29 2023 ->  1. You are running an out-of-date version of ClamAV / FreshClam.
Mon Mar  6 14:46:29 2023 ->     Ensure you are the most updated version by visiting https://www.clamav.net/downloads
Mon Mar  6 14:46:29 2023 ->  2. Your network is explicitly denied by the FreshClam CDN.
Mon Mar  6 14:46:29 2023 ->     In order to rectify this please check that you are:
Mon Mar  6 14:46:29 2023 ->    a. Running an up-to-date version of FreshClam
Mon Mar  6 14:46:29 2023 ->    b. Running FreshClam no more than once an hour
Mon Mar  6 14:46:29 2023 ->    c. If you have checked (a) and (b), please open a ticket at
Mon Mar  6 14:46:29 2023 ->       https://github.com/Cisco-Talos/clamav/issues
Mon Mar  6 14:46:29 2023 ->       and we will investigate why your network is blocked.
Mon Mar  6 14:46:29 2023 -> ^You are on cool-down until after: 2023-03-07 14:46:29
Mon Mar  6 14:46:29 2023 -> !Database update process failed: Forbidden; Blocked by CDN
Mon Mar  6 14:46:29 2023 -> !Update failed.

**Comment by micahsnyder** [comment]
> Since Cloudflare is in between my machine and the clamv servers: can it be that I'm on a Coudflare blacklist, or even the whole network is?

@mks-mr It's definitely possible. 

I don't have any other ideas for resolving the 403.  As a workaround, you could try using Microsoft's mirror of the ClamAV databases.  I learned that this exists just late January: https://packages.microsoft.com/clamav/  

It appears that they keep their mirror up-to-date but I can't make any promises.  I don't have any contacts with the people at Microsoft that host it.

**Comment by micahsnyder** [comment]
Since I believe this issue is out of our hands, I am going to close the ticket.  If you find some new information and you think we can help, feel free to re-open or open another issue.

**Comment by badlialrashid** [comment]
https://packages.microsoft.com/clamav/ 
I put it under /var/lib/clamav it did not get updated automatically. Use clamdscan it hangs after a couple of seconds.

**Comment by micahsnyder** [comment]
@badlialrashid Please share the output of `clamconf -n`.  It will show various configuration settings including where `clamd` and `clamscan` expect to find the signatures.

**Comment by badlialrashid** [comment]
Hi,

Thank you, it got updated.

Regards,
-badli

________________________________

**Comment by austinobioma** [comment]
How did you resolve it @badlialrashid ? I am having the same challenge at the moment

**Comment by badlialrashid** [comment]
HI Austin,

It suddenly got updated if I recall correctly because I put it the virusdb inside the database folder manually.

I think I did not use micahsnyder method.


Sent from Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by ZhakLilo** [comment]
Hi, 

We are experiencing the same issue on one of our servers. We get 403 response from Cloudflare:

`* About to connect() to database.clamav.net port 443 (#2)
*   Trying 104.16.218.84...
* Connected to database.clamav.net (104.16.218.84) port 443 (#2)
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
* SSL connection using TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
* Server certificate:
* 	subject: CN=sni.cloudflaressl.com,O="Cloudflare, Inc.",L=San Francisco,ST=California,C=US
* 	start date: May 15 00:00:00 2023 GMT
* 	expire date: May 14 23:59:59 2024 GMT
* 	common name: sni.cloudflaressl.com
* 	issuer: CN=Cloudflare Inc ECC CA-3,O="Cloudflare, Inc.",C=US
> GET /daily-26440.cdiff HTTP/1.1
User-Agent: ClamAV/0.102.4 (OS: linux-gnu, ARCH: x86_64, CPU: x86_64)
Host: database.clamav.net
Accept: */*
Connection: close

< HTTP/1.1 403 Forbidden
< Date: Wed, 06 Mar 2024 16:17:41 GMT
< Content-Type: text/html; charset=UTF-8
< Transfer-Encoding: chunked
< Connection: close
< X-Frame-Options: SAMEORIGIN
< Referrer-Policy: same-origin
< Cache-Control: max-age=15
< Expires: Wed, 06 Mar 2024 16:17:56 GMT
< Set-Cookie: __cf_bm=OkS_LqfWjqLgcGNrRBrLCUllKGd6.SrOSaMdVrCQHjM-1709741861-1.0.1.1-7F9CfFcbTb8ggy8GEwDoJIsMj_6wNtemPL1NgDhQZn3rHnj3m_leQYnKJduXTsA7Nai_ROKp26b34ak5bymG6g; path=/; expires=Wed, 06-Mar-24 16:47:41 GMT; domain=.clamav.net; HttpOnly; Secure; SameSite=None
< Strict-Transport-Security: max-age=15552000
< X-Content-Type-Options: nosniff
< Server: cloudflare
< CF-RAY: 8603a84a7b6b4942-TPA
<
* Closing connection 2
WARNING: downloadFile: Unexpected response (403) from https://database.clamav.net/daily-26440.cdiff
WARNING: getpatch: Can't download daily-26440.cdiff from https://database.clamav.net/daily-26440.cdiff
WARNING: Incremental update failed, trying to download daily.cvd
Retrieving https://database.clamav.net/daily.cvd
downloadFile: Download source:      https://database.clamav.net/daily.cvd
downloadFile: Download destination: /var/lib/clamav/tmp.3f2a1/clamav-da8b2528efc11a0b16574a28abfc8f4e.tmp`

Can you please check if we are blocked in some way? Any other suggestions are welcomed as well. Thank you in advance.

**Comment by micahsnyder** [comment]
@ZhakLilo can you check what the geo-IP results are for your public IP address using https://www.maxmind.com/en/geoip-demo ?  If it comes up as "Russia" then that would explain the block.  IP address blocks change hands relatively often across nation boundaries, so while I see your GitHub location reads "Bulgaria" it's likely that the IP address you're assigned is being identified as Russian.

**Comment by ZhakLilo** [comment]
Hi, actually if I visit directly from my computer (BG IP) the file is downloaded, and the server IP address is US Florida Tampa based, where I get this problem.

**Comment by micahsnyder** [comment]
@ZhakLilo  OOH I missed something obvious in your logs:

> User-Agent: ClamAV/0.102.4 (OS: linux-gnu, ARCH: x86_64, CPU: x86_64)

ClamAV 0.102.4 is past EOL and is blocked.  You need to upgrade to a newer version. 

https://docs.clamav.net/faq/faq-eol.html#version-support-matrix
